### PR TITLE
Add `derive serde` attributes to the generated message and service struct

### PR DESCRIPTION
**Pull request by ZhangHanDong**
hi @adnanademovic , 

thanks very much for your contribution ,  `rosrust` is a great crate. 

I recently plan to use it in production environment, but I  need to return `json` string directly from the generated struct.  But now , `rosrust` doesn't support this feature, so I  implemented it by myself, and  test available in the actual environment.  

I found that someone else has submitted the [PR #135](https://github.com/adnanademovic/rosrust/pull/135) of similar feature, but his is wrong and can't be compiled.

I hope you can merge and release a new crate version so that my project can use the new version instand of relying on git.

thanks !

---

usage:

```rust
rosrust = { version = "0.9.5",  features = ["derive-serde"]} 
```

### CODE REVIEW

**Comment by ZhangHanDong** [comment]
@adnanademovic

File: rosrust/Cargo.toml [hunk 1]
```diff
@@ -15,7 +15,7 @@ error-chain = "0.12.1"
 lazy_static = "1.4.0"
 log = "0.4.8"
 nix = "0.15.0"
-rosrust_codegen = "0.9.2"
+rosrust_codegen = 0.9.3
```
Comment (elihunter173) [inline rosrust/Cargo.toml:18]: I'm just some random person, but this should still be in quotes.

Comment (ZhangHanDong) [reply rosrust/Cargo.toml]: > I'm just some random person, but this should still be in quotes.

thanks

**Comment by adnanademovic** [comment]
Will this cause build failure if you try to serialize a fixed length array longer than 32?
That's the reason the feature isn't already there, due to const generics not existing yet, and thus serde has that limitation.

**Comment by ZhangHanDong** [comment]
The const generics function will be stable in version 1.51, and this defect may be improved in the future.

**Comment by adnanademovic** [comment]
Yeah once the feature is there, and serde implements it, adding serde support will be extremely simple, and I'd likely add it directly without any feature flags.

**Comment by ZhangHanDong** [comment]
ok

**Pull request closed without merging**